probe/docker/registry.go

```diff
@@ -176,7 +176,8 @@ func (r *registry) listenForEvents() bool {
 	// Next, start listening for events.  We do this before fetching
 	// the list of containers so we don't miss containers created
 	// after listing but before listening for events.
-	events := make(chan *docker_client.APIEvents)
+	// Use a buffered chan to reduce the chance of losing events
+	events := make(chan *docker_client.APIEvents, 32)
```

Comment:
Where do the events get dropped? Is the sender using a non-blocking send?

Reply:
Unclear. Not in `fsouza/go-dockerclient`.
Maybe the other side of the socket.

Reply:
Here you go: https://github.com/moby/moby/blob/a6e1502575858cd063b0e75d400e04b6f334e86d/daemon/events/events.go#L114

Reply:
Also `// Each listener has 100 milliseconds to receive the event or it will be skipped.`

Reply:
Is there any downside to making the buffer a lot larger (but not as large as to become a memory hog)?

Reply:
Like 1024?  I don't see any issue.
Really we need to do a full sync periodically, so I see this change as an improvement but not a final fix.

Reply:
1024 sounds reasonable.

Pls file an issue for the periodic sync.

Reply:
Filed https://github.com/weaveworks/scope/issues/3629